include/wsrep/xid.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2019 Codership Oy <info@codership.com>
+ *
+ * This file is part of wsrep-lib.
+ *
+ * Wsrep-lib is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Wsrep-lib is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with wsrep-lib.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef WSREP_XID_HPP
+#define WSREP_XID_HPP
+
+#include <ostream>
+
+namespace wsrep
+{
+    class xid
+    {
+    public:
+        xid() : xid_() { }
+        xid(const std::string& str) : xid_(str) { }
+        xid(const char* s) : xid_(s) { }
+        bool empty() const { return xid_.empty(); }
+        void clear() { xid_.clear(); }
+        bool operator==(const xid& other) const { return xid_ == other.xid_; }
+        friend std::string to_string(const wsrep::xid& xid);
+    private:
+        std::string xid_;
+    };
+
+    std::string to_string(const wsrep::xid& xid);
+    std::ostream& operator<<(std::ostream& os, const wsrep::xid& xid);
```

Comment:
`to_string(const wsrep::xid&)` does not seem to be used elsewhere than in `operator<<(std::ostream& os, const wsrep::xid& xid)`. It might make sense to make `operator<<(std::ostream& os, const wsrep::xid& xid)` a friend of `xid` and use `xid.xid_` there directly. This would avoid string copy each time `to_string(const wsrep::xid& xid)` is called from `operator<<`.

Reply:
I made `operator<<` a friend. However `to_string()` stays, because it's going to be used in full patch in mariadb/mysql side.